llvm/include/llvm/CodeGen/BasicTTIImpl.h

```diff
@@ -1186,8 +1186,10 @@ class BasicTTIImplBase : public TargetTransformInfoImplCRTPBase<T> {
 
       // Return the cost of multiple scalar invocation plus the cost of
       // inserting and extracting the values.
-      return getScalarizationOverhead(DstVTy, /*Insert*/ true, /*Extract*/ true,
-                                      CostKind) +
+      return getScalarizationOverhead(SrcVTy, /*Insert*/ false,
+                                      /*Extract*/ true, CostKind) +
+             getScalarizationOverhead(DstVTy, /*Insert*/ true,
+                                      /*Extract*/ false, CostKind) +
```

Comment:
```suggestion
      return getScalarizationOverhead(SrcVTy, /*Insert=*/ false,
                                      /*Extract=*/ true, CostKind) +
             getScalarizationOverhead(DstVTy, /*Insert=*/ true,
                                      /*Extract=*/ false, CostKind) +
```

Reply:
It doesn't really make sense to me that this would be an additive cost. For upcasts I'd expect to use the result type, and downcasts the source?

Reply:
This is specifically costing the case where we are scalarizing.  We're generating O(#lanes) scalar casts, and this part is just the insert/extract.